File: app/src/main/java/com/acdos/comp41690/ui/rain/RainViewModel.java
package com.acdos.comp41690.ui.rain;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class RainViewModel extends ViewModel {

    private MutableLiveData<String> mText;

    public RainViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is gallery fragment");
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/com/example/help/ui/rain/RainViewModel.kt
package com.example.help.ui.rain

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class RainViewModel : ViewModel() {

    private val _text = MutableLiveData<String>().apply {
        value = "This is rain Fragment"
    }
    val text: LiveData<String> = _text
}